ipyparallel/nbextension/clustermanager.py

```diff
@@ -0,0 +1,172 @@
+"""Manage IPython parallel clusters in the notebook."""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+
+from tornado import web
+
+from traitlets.config.configurable import LoggingConfigurable
+from traitlets import Dict, Instance, Float
+from ipython_genutils import py3compat
+
+
+class ClusterManager(LoggingConfigurable):
+
+    profiles = Dict()
+
+    delay = Float(1., config=True,
+        help="delay (in s) between starting the controller and the engines")
+
+    loop = Instance('zmq.eventloop.ioloop.IOLoop')
+    def _loop_default(self):
+        from zmq.eventloop.ioloop import IOLoop
+        return IOLoop.instance()
+
+    def build_launchers(self, profile_dir):
+        from ipyparallel.apps.ipclusterapp import IPClusterStart
+        
+        class DummyIPClusterStart(IPClusterStart):
+            """Dummy subclass to skip init steps that conflict with global app.
+    
+            Instantiating and initializing this class should result in fully configured
+            launchers, but no other side effects or state.
+            """
+
+            def init_signal(self):
+                pass
+            def reinit_logging(self):
+                pass
+        
+        starter = DummyIPClusterStart(log=self.log)
+        starter.initialize(['--profile-dir', profile_dir])
+        cl = starter.controller_launcher
+        esl = starter.engine_launcher
```

Comment:
This bit is unmodified from upstream, but it's because it's an "EngineSetLauncher"